tapa/common/graph.py

```diff
@@ -86,3 +93,101 @@ def get_flatten_graph(self) -> "Graph":
         }
 
         return Graph(self.name, new_obj)
+
+    def get_floorplan_slot(
+        self, slot_name: str, tasks_in_slot: list[str], top: Base
+    ) -> TaskDefinition:
+        """Return the task grouping the tasks floorplanned in the given slot."""
+        # Construct obj of the slot by modifying the top task
+        new_obj = self.get_top_task_def().to_dict()
+        new_obj["level"] = "upper"
+
+        # Find all task instances
+        insts = self.get_top_task_inst().get_subtasks_insts()
+        for inst in insts:
+            # make sure top has been flattened
+            assert isinstance(inst.definition, TaskDefinition)
+            assert inst.definition.get_level() == TaskDefinition.Level.LEAF, (
+                "Top task must be flattened for floorplanning"
+            )
+
+        # filter out tasks that are not in the slot
+        new_insts = [inst for inst in insts if inst.name in tasks_in_slot]
+
+        assert new_insts, [inst.name for inst in insts]
```

Comment:
Do you want to check that `all(inst in insts for inst in tasks_in_slot)`? It should be illegal to pass invalid instance names.

You might want to add unit tests to cover such cases.